src/object/class_object.c

```diff
@@ -8244,6 +8258,30 @@ classobj_check_index_exist (SM_CLASS_CONSTRAINT * constraints, char **out_shared
     case SM_SHARE_INDEX:
       if (out_shared_cons_name != NULL)
 	{
+#if defined(SUPPORT_DEDUPLICATE_KEY_MODE)
+	  if (constraint_type == DB_CONSTRAINT_FOREIGN_KEY)
+	    {
+	      int level;
+	      SM_ATTRIBUTE **attp = existing_con->attributes;
```

Comment:
assert (attp != NULL) is not needed?

Reply:
This is checked in the classobj_find_constraint_by_attrs() function performed earlier.
If existing_con exists, it must have an attribute.